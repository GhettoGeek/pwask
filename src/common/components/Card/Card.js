import React from 'react'
import PropTypes from 'prop-types'
import { useTranslation } from 'react-i18next'
import { useModal } from 'react-modal-hook'
import { makeStyles } from '@material-ui/core/styles'
import {
  Card, CardActions, CardContent, CardMedia, IconButton,
} from '@material-ui/core'
import {
  Favorite as FavoriteIcon,
  Share as ShareIcon,
  Directions as DirectionIcon,
} from '@material-ui/icons'
import SwipeableViews from 'react-swipeable-views'
import Typography from '../Typography'
import Pagination from '../Pagination'
import Link from '../Link'
import Dialog from '../Dialog'
import ShareButtons from '../ShareButtons'

const useStyles = makeStyles({
  images: {
    position: 'relative',
  },
  media: {
    height: 140,
  },
})

function EnhancedCard({
  id, title, description, images, getDirectionUrl,
}) {
  const { t } = useTranslation()
  const classes = useStyles()
  const [currentIndex, setCurrentIndex] = React.useState(0)
  const handleChangeIndex = (value) => setCurrentIndex(value)
  const [showShareDialog, closeShareDialog] = useModal(({ in: open = true }) => (
    <Dialog onClose={closeShareDialog} open={open}>
      <ShareButtons />
    </Dialog>
  ))

  return (
    <Card className={classes.card}>
      <div className={classes.images}>
        <SwipeableViews
          enableMouseEvents
          index={currentIndex}
          onChangeIndex={handleChangeIndex}
        >
          {images.map((image, index) => (
            <CardMedia
              key={index}
              image={image.url}
              title={image.title}
              className={classes.media}
            />
          ))}
        </SwipeableViews>
        <Pagination
          dots={images.length}
          currentIndex={currentIndex}
          onChangeIndex={handleChangeIndex}
        />
      </div>
      <CardContent>
        <Link to={`/resource/${id}`}>
          <Typography gutterBottom variant="h5" component="h2">
            {title}
          </Typography>
        </Link>
        <Typography variant="body1">
          {description.substring(0, 250)}
          ...
        </Typography>
      </CardContent>
      <CardActions>
        <IconButton
          aria-label={t('common.getDirection')}
          onClick={() => { window.location.href = getDirectionUrl }}
        >
          <DirectionIcon />
        </IconButton>
        <IconButton
          aria-label={t('common.addToFavorite')}
          onClick={() => {}}
        >
          <FavoriteIcon />
        </IconButton>
        <IconButton
          aria-label={t('common.share')}
          onClick={showShareDialog}
        >
          <ShareIcon />
        </IconButton>
      </CardActions>
    </Card>
  )
}

EnhancedCard.propTypes = {
  id: PropTypes.number.isRequired,
  title: PropTypes.string.isRequired,
  description: PropTypes.string,
  images: PropTypes.arrayOf(PropTypes.shape({
    url: PropTypes.string,
    title: PropTypes.string,
  })),
  getDirectionUrl: PropTypes.string.isRequired,
}

EnhancedCard.defaultProps = {
  description: '',
  images: [
    {
      label: 'default',
      src: '/images/placeholder.jpg',
    },
  ],
}

export default EnhancedCard
