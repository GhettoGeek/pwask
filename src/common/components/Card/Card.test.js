import React from 'react'
import { shallow } from 'enzyme'
import Card from './Card'

describe('Card', () => {
  it('renders correctly the component', () => {
    const wrapper = shallow(<Card />)

    expect(wrapper.html()).toMatchSnapshot()
  })
})
