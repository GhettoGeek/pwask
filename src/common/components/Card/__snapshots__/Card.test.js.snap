// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders correctly the component 1`] = `"<div class=\\"MuiPaper-root MuiPaper-elevation1 MuiCard-root MuiPaper-rounded\\"><div class=\\"makeStyles-images-1\\"><div style=\\"overflow-x:hidden\\"><div style=\\"-webkit-flex-direction:row;flex-direction:row;-webkit-transition:all 0s ease 0s;transition:all 0s ease 0s;direction:ltr;display:flex;will-change:transform\\" class=\\"react-swipeable-view-container\\"><div style=\\"width:100%;-webkit-flex-shrink:0;flex-shrink:0;overflow:auto\\" aria-hidden=\\"false\\" data-swipeable=\\"true\\"><div class=\\"MuiCardMedia-root makeStyles-media-2\\"></div></div></div></div></div><div class=\\"MuiCardContent-root\\"><a class=\\"MuiTypography-root MuiLink-root MuiLink-underlineNone MuiTypography-colorPrimary\\" href=\\"/resource/undefined\\"><h2 class=\\"MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom\\">Title</h2></a><p class=\\"MuiTypography-root MuiTypography-body1\\">...</p></div><div class=\\"MuiCardActions-root MuiCardActions-spacing\\"><button class=\\"MuiButtonBase-root MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\" aria-label=\\"common.getDirection\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M21.71 11.29l-9-9a.9959.9959 0 00-1.41 0l-9 9c-.39.39-.39 1.02 0 1.41l9 9c.39.39 1.02.39 1.41 0l9-9c.39-.38.39-1.01 0-1.41zM14 14.5V12h-4v3H8v-4c0-.55.45-1 1-1h5V7.5l3.5 3.5-3.5 3.5z\\"></path></svg></span></button><button class=\\"MuiButtonBase-root MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\" aria-label=\\"common.addToFavorite\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M12 21.35l-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z\\"></path></svg></span></button><button class=\\"MuiButtonBase-root MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\" aria-label=\\"common.share\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7s-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81 1.66 0 3-1.34 3-3s-1.34-3-3-3-3 1.34-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9c-1.66 0-3 1.34-3 3s1.34 3 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.16c-.05.21-.08.43-.08.65 0 1.61 1.31 2.92 2.92 2.92 1.61 0 2.92-1.31 2.92-2.92s-1.31-2.92-2.92-2.92z\\"></path></svg></span></button></div></div>"`;
